lib/ansible/modules/storage/netapp/na_ontap_qtree.py

```diff
@@ -110,107 +135,151 @@ def __init__(self):
             ],
             supports_check_mode=True
         )
-
-        p = self.module.params
-
-        # set up state variables
-        self.state = p['state']
-        self.name = p['name']
-        self.from_name = p['from_name']
-        self.flexvol_name = p['flexvol_name']
-        self.vserver = p['vserver']
+        self.na_helper = NetAppModule()
+        self.parameters = self.na_helper.set_parameters(self.module.params)
 
         if HAS_NETAPP_LIB is False:
             self.module.fail_json(
                 msg="the python NetApp-Lib module is required")
         else:
             self.server = netapp_utils.setup_na_ontap_zapi(
-                module=self.module, vserver=self.vserver)
+                module=self.module, vserver=self.parameters['vserver'])
 
     def get_qtree(self, name=None):
         """
         Checks if the qtree exists.
-
+        :param:
+            name : qtree name
         :return:
-            True if qtree found
+            Details about the qtree
             False if qtree is not found
         :rtype: bool
         """
         if name is None:
-            name = self.name
+            name = self.parameters['name']
 
         qtree_list_iter = netapp_utils.zapi.NaElement('qtree-list-iter')
         query_details = netapp_utils.zapi.NaElement.create_node_with_children(
-            'qtree-info', **{'vserver': self.vserver,
-                             'volume': self.flexvol_name,
+            'qtree-info', **{'vserver': self.parameters['vserver'],
+                             'volume': self.parameters['flexvol_name'],
                              'qtree': name})
-
         query = netapp_utils.zapi.NaElement('query')
         query.add_child_elem(query_details)
         qtree_list_iter.add_child_elem(query)
-
         result = self.server.invoke_successfully(qtree_list_iter,
                                                  enable_tunneling=True)
-
+        return_q = False
         if (result.get_child_by_name('num-records') and
                 int(result.get_child_content('num-records')) >= 1):
-            return True
-        else:
-            return False
+            return_q = {'export_policy': result['attributes-list']['qtree-info']['export-policy'],
```

Comment:
it may be safer to check that the key exists.   We have seen some changes between different ONTAP versions.  I can't say if this is an issue for qtree, but it may be safer to code defensively.